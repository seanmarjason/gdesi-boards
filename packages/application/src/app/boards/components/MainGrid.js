'use client'

import { useState } from 'react';

import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid2';
import Box from '@mui/material/Box';
import { Container } from '@mui/material';

import { DragDropContext } from '@hello-pangea/dnd';

import { Column } from './Column';

export default function MainGrid({ data=[] }) {

  const [columns, setColumns] = useState(data)

  let activeColumn = ''
  let activeCard = ''

  const handleColumnChange = (value) => {
    activeColumn = value.column
    activeCard = value.card
  };

  const handleOnDragEnd = () => {
    // Get card details
    const cards = columns.flatMap( column => column.cards)
    const activeCardDetails = cards.find(card => card.id == activeCard)

    // Move card
    setColumns(columns.map(column => { 
      // // remove card from old column
      column.cards = column.cards.filter(card => card.id != activeCard)
      // add card to new column
      if (column.id === activeColumn) {
        return {
            ...column,
            cards: [
              activeCardDetails,
              ...column.cards
            ] 
          };
      }
      return column;
      })
    )

  }

  return (
    <Box 
      // sx={{ 
      //   width: '100%', 
      //   maxWidth: { sm: '100%', md: '1700px' },
      //   height: '100%',
      //   }}
    >
      {/* columns */}
      <Typography component="h2" variant="h6" sx={{ mb: 2 }}>
        Your Team's Work
      </Typography>

      {console.log("columns", columns)}
      

      <DragDropContext
        onDragEnd={
          handleOnDragEnd
        }
      >

        <Container
          fixed
          sx={{
            display: 'flex',
            width: '100%',
            height: '100%',
            overflowX: 'scroll'
          }}
        >
            {
            columns ?
              columns.map((column, index) =>
                <Column
                  key={index}
                  column={column}
                  handleColumnChange={handleColumnChange}
                />
              )
              : <div></div>
            }
        </Container>

      </DragDropContext>


    </Box>
  );
}
