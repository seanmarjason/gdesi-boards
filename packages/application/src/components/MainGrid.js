'use client'

import { useState } from 'react';

import NextLink from 'next/link';

import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import { Container } from '@mui/material';
import Button from '@mui/material/Button';

import { DragDropContext } from '@hello-pangea/dnd';

import { Column } from './Column';

export default function MainGrid({ data, updateCardStatus, boardId }) {

  const [columns, setColumns] = useState(data)

  let activeColumn = ''
  let activeCard = ''

  const handleColumnChange = (value) => {
    activeColumn = value.column
    activeCard = value.card
  };

  const handleOnDragEnd = () => {
    // Get card details
    const cards = columns.flatMap( column => column.cards)
    const activeCardDetails = cards.find(card => card.id == activeCard)

    updateCardStatus(activeCard, columns[activeColumn - 1].title).then(
      // Move card
      setColumns(columns.map(column => { 
        // // remove card from old column
        column.cards = column.cards.filter(card => card.id != activeCard)
        // add card to new column
        if (column.id === activeColumn) {
          return {
              ...column,
              cards: [
                activeCardDetails,
                ...column.cards
              ] 
            };
        }
        return column;
        })
      )
    )
  }

  return (
    <Box sx={{
      width: '100%'
    }}>
      <Typography component="h2" variant="h6" sx={{ mb: 2 }}>
        Your Team's Work
      </Typography>

      <Button component={NextLink} href={`/boards/${boardId}/tasks/new`} variant="contained" color="secondary">
        Create New Task
      </Button>

      <DragDropContext
        onDragEnd={
          handleOnDragEnd
        }
      >

        <Container
          fixed
          sx={{
            display: 'flex',
            width: '100%',
            height: '100%',
            overflowX: 'scroll'
          }}
        >
            {
            columns ?
              columns.map((column, index) =>
                <Column
                  key={index}
                  column={column}
                  handleColumnChange={handleColumnChange}
                />
              )
              : <div></div>
            }
        </Container>

      </DragDropContext>


    </Box>
  );
}
