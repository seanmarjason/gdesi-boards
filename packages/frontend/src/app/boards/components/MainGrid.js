import * as React from 'react';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid2';
import Box from '@mui/material/Box';

export default function MainGrid() {
  return (
    <Box sx={{ width: '100%', maxWidth: { sm: '100%', md: '1700px' } }}>
      {/* cards */}
      <Typography component="h2" variant="h6" sx={{ mb: 2 }}>
        Your Team's Work
      </Typography>

      <Grid
        container
        spacing={2}
        columns={12}
        sx={{ mb: (theme) => theme.spacing(2) }}
      >


      </Grid>


    </Box>
  );
}
