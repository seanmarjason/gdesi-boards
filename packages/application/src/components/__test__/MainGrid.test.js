import { render } from '@testing-library/react'
import MainGrid from '../MainGrid';

jest.mock("@hello-pangea/dnd");

global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(''),
    }),
  );
  
 
const data = [
    {id: 1, name: 'To Do', cards: []},
    {id: 2, name: 'Doing', cards: []},
    {id: 3, name: 'Done', cards: []},
]

test('MainGrid renders correctly', () => {
  const { container } = render(<MainGrid data={data} updateCardStatus={jest.fn()} boardId={1}  />)
  expect(container).toMatchSnapshot()
})
