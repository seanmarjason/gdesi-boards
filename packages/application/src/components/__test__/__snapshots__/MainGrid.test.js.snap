// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainGrid renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h6 css-1hd3wng-MuiTypography-root"
    >
      Your Team's Work
    </h2>
    <a
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mns5ui-MuiButtonBase-root-MuiButton-root"
      href="/boards/1/tasks/new"
      id=":r0:"
      tabindex="0"
    >
      Create New Task
    </a>
  </div>
</div>
`;
